Server/src/main/java/org/openas2/processor/msgtracking/DbTrackingModule.java

```diff
@@ -278,23 +274,23 @@ public HashMap<String, String> showMessage(String msg_id) {
     }
 
     public ArrayList<HashMap<String, String>> getDataCharts(HashMap<String, String> map) {
-
         Connection conn = null;
         ArrayList<HashMap<String, String>> rows = new ArrayList<HashMap<String, String>>();
-
         try {
             if (useEmbeddedDB) {
                 conn = dbHandler.getConnection();
             } else {
                 conn = DriverManager.getConnection(jdbcConnectString, dbUser, dbPwd);
             }
+            CallableStatement cs = conn.prepareCall("{CALL ftGetMessagesByDateRange(?,?)}");
+            cs.setDate(1, Date.valueOf(map.get("startDate")));
+            cs.setDate(2,Date.valueOf(map.get("endDate")));
+            ResultSet rs = cs.executeQuery();
```

Comment:
This is your own custom callable.
It will not be added to OpenAS2 since the goal is to remain generic.
I assume the statement that is currently used does not work on SqlServer because it does not support the TIMESTAMP operator so instead change the statement to this and it will work on all SQL-92 and newer standards compliant database servers:
```
            ResultSet rs = s.executeQuery("SELECT " + FIELDS.MSG_ID + ",STATE,STATUS,CREATE_DT FROM " + tableName
                    + " WHERE CREATE_DT BETWEEN CAST('" + map.get("startDate").toString()
                    + " 00:00:00' as TIMESTAMP) AND CAST('" + map.get("endDate").toString()
                    + " 23:59:59'" as TIMESTAMP));
```